src/envoy/http/authn/authn_utils.cc

```diff
@@ -91,11 +114,18 @@ bool AuthnUtils::GetJWTPayloadFromHeaders(
     auto json_obj = Json::Factory::loadFromString(payload_str);
     ENVOY_LOG(debug, "{}: json object is {}", __FUNCTION__,
               json_obj->asJsonString());
+    // Extract claims
     json_obj->iterate(
         [payload](const std::string& key, const Json::Object& obj) -> bool {
-          return JsonIterCallback::JsonIteratorCallback(key, obj, payload);
+          return JsonIterCallback::JsonIteratorCallbackForJwtClaims(key, obj,
+                                                                    payload);
+        });
+    // Extract audience
+    json_obj->iterate(
```

Comment:
Why do you need to iterate for aud? Just try getStringArray or getString with key = 'aud' directly. You can also check the claims map for "aud" first. If found, just use it as single aud, otherwise try getStringArray. So that, don't need to try getString for aud.

Reply:
Done.